app/src/main/java/org/simple/clinic/instantsearch/InstantSearchEffectHandler.kt

```diff
@@ -73,6 +77,29 @@ class InstantSearchEffectHandler @AssistedInject constructor(
     }
   }
 
+  private fun checkIfPatientAlreadyHasAnExistingNHID(scheduler: Scheduler): ObservableTransformer<CheckIfPatientAlreadyHasAnExistingNHID, InstantSearchEvent> {
+    return ObservableTransformer { effects ->
+      effects
+          .observeOn(schedulers.computation())
+          .flatMap { patientRepository.patientProfile(it.patientId).subscribeOn(scheduler) }
+          .map { patientProfile ->
+            val patientUUID = patientProfile.get().patientUuid
```

Comment:
We can use an additional `.map` operator to move the unwrapping of the `Optional` outside this block.